src/views/interface/checkovResult/webviewPanel.ts

```diff
@@ -147,6 +150,33 @@ export class CheckovResultWebviewPanel {
         return htmlTemplate;
     }
 
+    public static async fetchDescription(checkId: string): Promise<string | undefined> {
+        let retryCount = 0;
+        const jwtToken = AnalyticsService.applicationContext.globalState.get(GLOBAL_CONTEXT.JWT_TOKEN) as string;
+
+        try {
+            const response = await axios.get(CheckovResultWebviewPanel.getDescriptionsEndpoint(checkId), { headers: {
+                'Authorization': jwtToken } });
+            
+                if (response.status === 200) {
+                    return response.data.description;
+                }
+        } catch (e: any) {
+            if (retryCount === 3) {
+                throw new Error('Unable to fetch description: ' + e.message);
+            }
+
+            if (e.response.status === 403) {
+                retryCount++;
+                await AnalyticsService.setAnalyticsJwtToken();
+                return await CheckovResultWebviewPanel.fetchDescription(checkId);
```

Comment:
Isn't it infinite recursion, shouldn't we pass the `retryCount` as a parameter? I think the `retryCount` will be always set to 0 on call.

Reply:
agree

Reply:
fixed